### DISCUSSION TITLE
How to properly sync shared tasks for collaborative use

### DISCUSSION THREAD

**Discussion by akuntsch** [TechbeeAT/jtxBoard#1629] [Q&A]
Hi, great app you got here but I'm having problems using this as a collaborative todo list.

I've set up a shared calender with VTODO support (in Nextcloud via DAVx5, if that's relevant) and I am now trying to get tasks and subtasks to sync more or less immediately when refreshing/forcing sync in the jtxBoard app.

Whenever I make changes (e.g. add subtasks), they are not synced to another device when refreshing in jtxBoard.
It only syncs correctly when forcing a sync in DAVx5.

Am I doing something wrong here?

I already checked https://jtx.techbee.at/troubles-with-the-synchronization-see-what-could-go-wrong and have configured everything correctly.

**Comment by akuntsch** [comment]
@patrickunterwegs sorry for the bump, but any ideas?

**Reply by patrickunterwegs** [reply]
Hi @akuntsch , sorry for the late reply. 
That's a bit hard to say. Basically how it works: 
Android has a sync framework in the background, this can trigger sync code one-time or regularly in another app.  jtx Board itself does not have a scheduled sync and does not trigger the sync in DAVx5, but only tells the sync framework that a sync was requested. 
DAVx5 on the other hand listens to the sync framework and starts the synchronization process (additionally DAVx5 has a scheduler that triggers the sync regularly). 

If the sync basically works when you start it out of DAVx5, there's something wrong with the integration of the sync framework and DAVx5. This could be because of battery optimization issues or any other block of the sync framework handing over the information to DAVx5. My recommended next step would be to check if there's any battery optimizations on or if you can find anything suspicious in the permissions. As far as I know older devices can also disable the sync which might also cause this problem. 

If all that does not help, please also consider checking with DAVx5 as this problem is actually out of scope of jtx Board (because of the technical reason above).
